bundles/binding/org.openhab.binding.tacmi/src/main/java/org/openhab/binding/tacmi/internal/message/AnalogValue.java

```diff
@@ -23,6 +25,7 @@
 public final class AnalogValue {
     public BigDecimal value;
     public TACmiMeasureType measureType;
+    public final MathContext mc = new MathContext(6, RoundingMode.HALF_UP);
```

Comment:
You are right, my first fix was quite dirty. The new version now really does what it should and does not introduce any precision that doesn't exist in reality.